### PR TITLE
BAC-20: Rate limiting middlware

### PR DESCRIPTION
**Pull request by binayachaudari**

### CODE REVIEW

**Review by Cyantosh0 — APPROVED** [review]
Thank you!

**Comment by Cyantosh0** [comment]
@binayachaudari sir, Could you please set the `period` and `limit` values in the default configuration so that we can just add a rate limiter handler without passing period and limit in every route or routes group.
If an endpoint needs a different `period` and `limit` then let's update the values through the route. You can take the reference from upload middleware.

File: api/routes/user_routes.go [hunk 1]
```diff
@@ -33,14 +35,16 @@ func NewUserRoutes(
...
 	}
 }
 
 // Setup user routes
 func (s *UserRoutes) Setup() {
 	s.logger.Info("Setting up routes")
 
-	api := s.handler.Group("/api").Use(s.authMiddleware.HandleAuthWithRole(constants.RoleIsAdmin))
+	api := s.handler.Group("/api").Use(s.authMiddleware.HandleAuthWithRole(constants.RoleIsAdmin),
+		s.rateLimitMiddleware.Handle(nil))
```
Comment (Cyantosh0) [inline api/routes/user_routes.go]: @binayachaudari Passing `nil` here seems odd. Could you please check similar way to update the options as similar in upload middleware config.

**Review by Cyantosh0 — COMMENTED** [review]
@binayachaudari Please check the comment.

File: api/routes/user_routes.go (hunk 1 above)
Comment (binayachaudari) [reply api/routes/user_routes.go]: Using `variadic` function to pass `period` and `limit` values. 
Commit Hash: [1fce82d](https://github.com/wesionaryTEAM/go_clean_architecture/pull/73/commits/1fce82d6e1cbc62585b108e0377852400f2e7f61)



Please verify.

**Review by Cyantosh0 — APPROVED** [review]
Thank you @binayachaudari sir

**Pull request merged**